src/new_recur.jl

```diff
@@ -0,0 +1,105 @@
+
+
+
+##### Helper scan funtion which can likely be put into NNLib. #####
+"""
+  scan
+
+Recreating jax.lax.scan functionality in julia.
+"""
+function scan_full(func, init_carry, xs::AbstractVector{<:AbstractArray})
+  # xs = Flux.eachlastdim(x_block)
+  x_init, x_rest = Iterators.peel(xs)
```

Comment:
We use x_rest in the foldl call to go through the rest of the sequence.

The reason peel was chosen here was not efficiency, but oddly gradient related. If instead I did
```
x_init = first(xs)
x_rest = xs[begin+1:end]
```

the resulting gradients were wrong. I've since documented this in the function.

Reply:
That seems very concerning. Do you have a MWE?

Reply:
No, no mwe. Commenting out the peel and replacing with the code block, I haven't been able to figure out where the gradients are going wrong here. Only know they fail.